packages/react-instantsearch/src/components/classNames.js

```diff
@@ -2,11 +2,18 @@ import cx from 'classnames';
 const prefix = 'ais';
 
 export default function classNames(block) {
-  return (...elements) => ({
-    className: cx(
-      elements
-        .filter(element => element !== undefined && element !== false)
-        .map(element => `${prefix}-${block}__${element}`)
-    ),
-  });
+  return (...bemElements) =>
+    cx(
+      bemElements
+        .filter(
+          element =>
+            element !== undefined && element !== false && element !== null
+        )
+        .map(
+          element =>
+            element === ''
+              ? `${prefix}-${block}`
+              : `${prefix}-${block}-${element}`
+        )
+    );
```

Comment:
Not sure about what you mean in your example. We need to be able to either return `ais-Widget` or `ais-Widget-element`. With your example it would just return `-element` or an empty string?

Reply:
Nope it will have the same behaviour, if the element is defined then it will return `ais-Widget-element` otherwise it returns `ais-Widget`. Keep as it is if you think it's simpler.

Reply:
Slightly different behaviour (the class is doubled for some reason). Let's keep it this way for now if you don't mind.